modules/core/src/main/scala/com.snowplowanalytics.snowplow/scalatracker/Emitter.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2015-2018 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.scalatracker
+
+import java.util.UUID
+
+/**
+ * This trait includes two methods unrelated to actual sending: `getCurrentMilliseconds` and `generateUUID`.
+ * They are needed here because `Tracker` is referentially transparent, and we leave it up for the emitters
+ * to decide whether `F` will be referentially transparent also.
+ */
+trait EmitterAlgebra[F[_]] {
```

Comment:
ah sorry, I meant one algebra for ms and one for uuid since they're not related.

Reply:
one `Clock[F[_]]` and one `UUID[F[_]]` for example